### PR TITLE
DSD-1397: Removing Header/Footer and updating dev guide

### PR DESCRIPTION
**Pull request by EdwinGuzman**
Fixes JIRA ticket [DSD-1397](https://jira.nypl.org/browse/DSD-1397)

## This PR does the following:

- This is a PR update from #1293  which is now closed and was pointing to the `development` branch.
- Removes the DS `Header` and `Footer` components and all the previous relevant pages.
- Keeps documentation for the components under the "Development Guide" Storybook section.

## How has this been tested?

Locally.

## Accessibility concerns or updates

- N/A

### Checklist:

- [ ] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [ ] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

File: src/components/DevelopmentGuide/Header.mdx [hunk 1]
```diff
@@ -1,71 +1,68 @@
...
-
-<ArgsTable story="Header" />
+## Implementation
+
+The `Header` can be embedded on an NYPL app through the following code snippet:
+
+<Source
+  code={`
+<style>
+  #Header-Placeholder {
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/Header.mdx]: Is there a reason for using capital letters?

File: src/components/DevelopmentGuide/Header.mdx [hunk 2]
```diff
@@ -1,71 +1,68 @@
...
-<Canvas withToolbar>
-  <Story name="Header" args={{ isProduction: true }}>
-    {({ args }) => <Header {...args} />}
-  </Story>
-</Canvas>
-
-<ArgsTable story="Header" />
+## Implementation
+
+The `Header` can be embedded on an NYPL app through the following code snippet:
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/Header.mdx]: Add something to explain why this is important (the fixed height styles).  Also, I think `can` should be changed to `should`.

File: src/components/DevelopmentGuide/Header.mdx [hunk 3]
```diff
@@ -1,71 +1,68 @@
...
 - [Navigation](#navigation)
 - [Header Log In Timeout](#header-log-in-timeout)
-- [Google Analytics](#google-analytics)
 
 ## Overview
 
-<Description of={Header} />
+The NYPL `Header` is the top most feature on NYPL sites. Even though this component
+appears on various NYPL digital applications, it is itself a separate application
+that gets embedded on said digital applications. This complex component implements
```
Comment (bigfishdesign13) [inline src/components/DevelopmentGuide/Header.mdx]: Do you think `is composed using` would be better than `implements`?

**Review by bigfishdesign13 — CHANGES_REQUESTED** [review]
Some minor changes.

File: src/components/DevelopmentGuide/Header.mdx (hunk 1 above)
Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/Header.mdx]: Legacy code.

Comment (bigfishdesign13) [reply src/components/DevelopmentGuide/Header.mdx]: For consistency, flip that to all lowercase.

Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/Header.mdx]: It's from the Header app repo so that will have to be updated as well.

Comment (EdwinGuzman) [reply src/components/DevelopmentGuide/Header.mdx]: We should also revisit this in the next phase of Header updates.

Comment (bigfishdesign13) [reply src/components/DevelopmentGuide/Header.mdx]: Got it.  Sounds good.

**Review by bigfishdesign13 — APPROVED** [review]
Looks good!

**Pull request merged**